processing/src/main/java/io/druid/segment/Metadata.java

```diff
@@ -137,19 +164,25 @@ public boolean equals(Object o)
 
     Metadata metadata = (Metadata) o;
 
-    if (!container.equals(metadata.container)) {
+    if (container != null ? !container.equals(metadata.container) : metadata.container != null) {
       return false;
     }
     // Probably incorrect - comparing Object[] arrays with Arrays.equals
```

Comment:
I usually use the "java.util.Objects.equals and hashCode (java 7+)" option since the generated code is cleaner